chainmanager/types/params.go

```diff
@@ -0,0 +1,148 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/maticnetwork/heimdall/params/subspace"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+)
+
+// Default parameter values
+const (
+	DefaultTxConfirmationTime time.Duration = 6 * 14 * time.Second
+)
+
+// Parameter keys
+var (
+	KeyTxConfirmationTime = []byte("TxConfirmationTime")
+	KeyChainParams        = []byte("ChainParams")
+)
+
+var _ subspace.ParamSet = &Params{}
+
+// ChainParams chain related params
+type ChainParams struct {
+	BorChainID            string                  `json:"bor_chain_id" yaml:"bor_chain_id"`
+	MaticTokenAddress     hmTypes.HeimdallAddress `json:"matic_token_address" yaml:"matic_token_address"`
+	StakingManagerAddress hmTypes.HeimdallAddress `json:"staking_manager_address" yaml:"staking_manager_address"`
+	RootChainAddress      hmTypes.HeimdallAddress `json:"root_chain_address" yaml:"root_chain_address"`
+	StakingInfoAddress    hmTypes.HeimdallAddress `json:"staking_info_address" yaml:"staking_info_address"`
+	StateSenderAddress    hmTypes.HeimdallAddress `json:"state_sender_address" yaml:"state_sender_address"`
+
+	// Bor Chain Contracts
+	StateReceiverAddress hmTypes.HeimdallAddress `json:"state_receiver_address" yaml:"state_receiver_address"`
+	ValidatorSetAddress  hmTypes.HeimdallAddress `json:"validator_set_address" yaml:"validator_set_address"`
```

Comment:
Can we have a map instead? So we can add more contracts if needed in the future?

Reply:
@vaibhavchellani  Even if we add `map` we need to change the Heimdall and upgrade the system to use that address.

Reply:
We will just a tx which adds another key to the map

Reply:
Yeah, that is okay but to use that key we need to change the code somewhere. So it won't be a dynamic upgrade.

Reply:
True. It's just a step in that direction. Also allows us to iterate through all the contracts that are being used to do certain operations like `validateHeimdallAddress`